prow/crier/controller.go

```diff
@@ -160,24 +175,40 @@ func (c *Controller) processNextItem() bool {
 			return true
 		}
 
-		if c.queue.NumRequeues(key) < 5 {
-			logrus.WithError(err).WithField("prowjob", keyRaw).Error("Failed processing item, retrying")
-			c.queue.AddRateLimited(key)
-		} else {
-			logrus.WithError(err).WithField("prowjob", keyRaw).Error("Failed processing item, no more retries")
-			c.queue.Forget(key)
-			utilruntime.HandleError(err)
+		return retry(c.queue, key, err)
+	}
+
+	// initiate PrevReportState, if not exist
+	if pj.Status.PrevReportState == nil {
```

Comment:
Why not just enforce this is non-nil on creation?

Reply:
ack